src/nasher/utils/git.nim

```diff
@@ -1,12 +1,15 @@
-import os, osproc, strutils, uri
+import os, osproc, strformat, strutils, uri
+import cli
 
 from shared import withDir
 
 proc gitExecCmd(cmd: string, default = ""): string =
   ## Runs ``cmd``, returning its output on success or ``default`` on error.
+  ## If default is empty, returns error text.
   let (output, errCode) = execCmdEx(cmd)
   if errCode != 0:
-    default
+    if default.len == 0: output.strip
```

Comment:
Restored previous function and added a carrier for the last error, if there was one, since that's what I was looking for with that change.  Not sure if the methodology is "good" programming in nim, so feel free to tell me to find another way if you don't like it.